### PR TITLE
added [[maybe_unused]] in some places where it was needed

### PR DESCRIPTION
**Pull request by 1mi0**
- [*] My pull request is made against the `dev` branch.
- [*] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [*] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [*] I tested my commits, by adding a test case to the unit tests if needed
- [*] I have ensured that I did not break any existing API calls.
- [?] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [*] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.

### CODE REVIEW

**Comment by braindigitalis** [comment]
thanks! please retarget this pr against the dev branch

File: include/dpp/json_interface.h [hunk 1]
```diff
@@ -44,7 +44,7 @@ namespace dpp {
 		 * @param j nlohmann::json object
 		 * @return T& Reference to self for fluent calling
 		 */
-		T& fill_from_json(nlohmann::json* j) {
+		T& fill_from_json([[maybe_unused]] nlohmann::json* j) {
```
Comment (braindigitalis) [inline include/dpp/json_interface.h:47]: how is this ever unused? we always pass json in.

File: include/dpp/json_interface.h [hunk 2]
```diff
@@ -54,7 +54,7 @@ namespace dpp {
 		 * @param with_id Include the ID in the JSON
 		 * @return std::string JSON string version of object
 		 */
-		virtual std::string build_json(bool with_id = false) const {
+		virtual std::string build_json([[maybe_unused]] bool with_id = false) const {
```
Comment (braindigitalis) [inline include/dpp/json_interface.h:57]: how can this be maybe unused when it's an optional parameter?

**Review by braindigitalis — CHANGES_REQUESTED** [review]

File: include/dpp/timed_listener.h [hunk 3]
```diff
@@ -73,7 +73,7 @@ template <typename attached_event, class listening_function> class timed_listene
 		/* Attach event */
 		listener_handle = ev(listener);
 		/* Create timer */
-		th = cl->start_timer([this](dpp::timer timer_handle) {
+		th = cl->start_timer([this]([[maybe_unused]] dpp::timer timer_handle) {
```
Comment (Jaskowicz1) [inline include/dpp/timed_listener.h:76]: Why is this "maybe unused"? When would you use a timer without a lambda or storing the timer 🤔

**Comment by 1mi0** [comment]
To answer all questions:
[[maybe_unused]] in our situation tells the compiler that we don't use the parameters inside of their functions. Not that we won't use them when we call the function or in other overloads | overrides of the function. In these specific 4 cases it just so happens that those arguments are not used inside of their lambda/function causing unused variable errors in my instance and warnings in other instances.

File: include/dpp/json_interface.h (hunk 1 above)
Comment (Mishura4) [reply include/dpp/json_interface.h:47]: It's unused in this definition, causing a warning

File: include/dpp/json_interface.h (hunk 2 above)
Comment (Mishura4) [reply include/dpp/json_interface.h:57]: Same as my other comment

**Review by Mishura4 — APPROVED** [review]

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**